products/BellHybrid/apps/include/AudioApplication.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <Application.hpp>
+#include <common/models/AbstractAudioModel.hpp>
+#include <common/models/AudioModel.hpp>
+
+namespace app
+{
+    class AudioApplication : public Application
+    {
+      protected:
+        std::shared_ptr<AbstractAudioModel> audioModel;
+
+      public:
+        AudioApplication(std::string name,
+                         std::string parent                  = "",
+                         StatusIndicators statusIndicators   = StatusIndicators{},
+                         StartInBackground startInBackground = {false},
+                         uint32_t stackDepth                 = 4096,
+                         sys::ServicePriority priority       = sys::ServicePriority::Idle)
+            : Application(std::move(name), std::move(parent), statusIndicators, startInBackground, stackDepth),
+              audioModel{std::make_shared<AudioModel>(this)}
+        {
+            bus.channels.push_back(sys::BusChannel::ServiceAudioNotifications);
+        }
+    };
+} // namespace app
```

Comment:
I get you are trying to remove unnecessary code duplication, which is fine but I'm not sure, if this is the correct solution. I strongly believe, the composition should be used in favor of inheritance. After all, it's just two lines more that needs to be added to applications:
```
std::unique_ptr<AbstractAudioModel> audioModel;
bus.channels.push_back(sys::BusChannel::ServiceAudioNotifications);
```